src/components/vm/consoles/VncConsole.jsx

```diff
@@ -0,0 +1,155 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+ */
+
+/* This started as a copy of the VncConsole component of @patternfly/react-console.
+
+   https://github.com/patternfly/react-console/blob/main/packages/module/src/components/VncConsole/VncConsole.tsx
+
+   It was stripped down to only contain the raw RFB widget from novnc.
+   Actions, CSS, and handling of the connected/disconnected state was
+   removed.
+ */
+
+import React from 'react';
+
+import { initLogging } from '@novnc/novnc/lib/util/logging';
+import RFB_module from '@novnc/novnc/lib/rfb';
+const RFB = RFB_module.default;
+
+export const VncConsole = ({
+    children,
+    host,
+    port = '80',
+    path = '',
+    encrypt = false,
+    resizeSession = true,
+    clipViewport = false,
+    dragViewport = false,
+    scaleViewport = false,
+    viewOnly = false,
+    shared = false,
+    credentials,
+    repeaterID = '',
+    vncLogging = 'warn',
+    consoleContainerId,
+    onDisconnected = () => {},
+    onInitFailed,
+    onSecurityFailure,
+}) => {
+    const rfb = React.useRef();
+
+    const novncElem = React.useRef(null);
+
+    const _onDisconnected = React.useCallback(
+        (e) => {
+            onDisconnected(e);
+        },
+        [onDisconnected]
+    );
+
+    const _onSecurityFailure = React.useCallback(
+        (e) => {
+            onSecurityFailure(e);
+        },
+        [onSecurityFailure]
+    );
+
+    const addEventListeners = React.useCallback(() => {
+        if (rfb.current) {
+            rfb.current?.addEventListener('disconnect', _onDisconnected);
+            rfb.current?.addEventListener('securityfailure', _onSecurityFailure);
+        }
+    }, [rfb, _onDisconnected, _onSecurityFailure]);
+
+    const removeEventListeners = React.useCallback(() => {
+        if (rfb.current) {
+            rfb.current.removeEventListener('disconnect', _onDisconnected);
+            rfb.current.removeEventListener('securityfailure', _onSecurityFailure);
+        }
+    }, [rfb, _onDisconnected, _onSecurityFailure]);
+
+    const connect = React.useCallback(() => {
+        const protocol = encrypt ? 'wss' : 'ws';
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2008-7727bbc3-20250715-165445-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)